src/CoreWCF.Kafka/src/CoreWCF/Channels/KafkaTransportPump.cs

```diff
@@ -0,0 +1,230 @@
+    private static readonly (bool? EnableAutoCommit, bool? EnableAutoOffsetStore) s_atLeastOnceBatchCommitConfigValues = (true, false);
+
+    public KafkaTransportPump(KafkaTransportBindingElement transportBindingElement,
+        KafkaSecurityBindingElement securityBindingElement,
+        ILogger<KafkaTransportPump> logger,
+        IServiceDispatcher serviceDispatcher, KafkaDeliverySemantics kafkaDeliverySemantics)
+    {
+        _logger = logger;
+        _kafkaDeliverySemantics = kafkaDeliverySemantics;
+        _topic = serviceDispatcher.BaseAddress.PathAndQuery.TrimStart('/');
+        _transportBindingElement = transportBindingElement;
+        _securityBindingElement = securityBindingElement;
+        _baseAddress = serviceDispatcher.BaseAddress;
+    }
+
+    private ConsumerConfig BuildConfig()
+    {
+        var config = new ConsumerConfig
+        {
+            BootstrapServers = _baseAddress.Authority,
+            IsolationLevel = IsolationLevel.ReadCommitted,
+            AutoOffsetReset = AutoOffsetReset.Earliest
+        };
+
+        foreach (var keyValue in _transportBindingElement.Config)
+        {
+            config.Set(keyValue.Key, keyValue.Value);
+        }
+
+        foreach (var keyValue in _securityBindingElement.Config)
+        {
+            config.Set(keyValue.Key, keyValue.Value);
+        }
+
+
+        var (enableAutoCommit, enableAutoOffsetStore) = GetCommitStrategyConfigValues(config, _kafkaDeliverySemantics);
+        config.EnableAutoCommit = enableAutoCommit;
+        config.EnableAutoOffsetStore = enableAutoOffsetStore;
+
+        return config;
+    }
+
+    private static (bool? EnableAutoCommit, bool? EnableAutoOffsetStore) GetCommitStrategyConfigValues(ConsumerConfig consumerConfig, KafkaDeliverySemantics kafkaDeliverySemantics) =>
+        (kafkaDeliverySemantics, consumerConfig) switch
+        {
+            // KafkaBinding
+            (KafkaDeliverySemantics.AtMostOnce, { EnableAutoCommit: null, EnableAutoOffsetStore: null }) => s_atMostOnceConfigValues,
+            (KafkaDeliverySemantics.AtLeastOnce, { EnableAutoCommit: null, EnableAutoOffsetStore: null } ) => s_atLeastOnceBatchCommitConfigValues,
+            // CustomBinding
+            (KafkaDeliverySemantics.AtMostOnce, { EnableAutoCommit: false, EnableAutoOffsetStore: null }) => s_atMostOnceConfigValues,
+            (KafkaDeliverySemantics.AtLeastOnce, { EnableAutoCommit: true, EnableAutoOffsetStore: false } ) => s_atLeastOnceBatchCommitConfigValues,
+            (KafkaDeliverySemantics.AtLeastOnce, { EnableAutoCommit: false, EnableAutoOffsetStore: null }) => s_atLeastOncePerMessageCommitConfigValues,
+            _ => throw new NotSupportedException(string.Format(SR.ProvidedKafkaConfigurationIsNotValid, kafkaDeliverySemantics, consumerConfig.EnableAutoCommit, consumerConfig.EnableAutoOffsetStore))
```

Comment:
can we do this validation before we start service ?

Reply:
It's difficult to do before service start as there are scenarios where EnableAutoOffsetStore can be null, and there are scenarios where EnableAutoCommit can be false. If you tried to validate before starting, where would you put the validation? It can't be any property setter as you can't presume order of properties being set and it's only the combination of values which is a problem. As long as the exception is thrown during startup, that's fine.